x/stablestake/keeper/msg_server_unbond.go

```diff
@@ -41,7 +42,13 @@ func (k msgServer) Unbond(goCtx context.Context, msg *types.MsgUnbond) (*types.M
 	}
 
 	redemptionAmount := sdk.NewDecFromInt(shareCoin.Amount).Mul(params.RedemptionRate).RoundInt()
-	redemptionCoin := sdk.NewCoin(params.DepositDenom, redemptionAmount)
+	entry, found := k.assetProfileKeeper.GetEntry(ctx, params.DepositDenom)
+	if !found {
+		return nil, errors.New("invalid denom")
```

Comment:
Can you register error and use registered error?

Reply:
Okay no problem.